addons/website/static/src/snippets/s_countdown/000.js

```diff
@@ -205,6 +205,12 @@ const CountdownWidget = publicWidget.Widget.extend({
                 const ctx = canvas.getContext("2d");
                 ctx.canvas.width = this.width;
                 ctx.canvas.height = this.size;
+                let maxWidth = 100 / this.display.length;
+                // due to mx class applied on the boxes layout, we could not use 33.33% or 25%
+                if (this.layout === 'boxes') {
+                    maxWidth -= this.display.length === 3 ? 7 : 5;
+                }
+                ctx.canvas.style.maxWidth = `${maxWidth}%`;
```

Comment:
Instead of playing with max widths and random percentages depending on the layout, can't we just put the canvas wrapper in flex display forced on one row so that the canvas are automatically shrunk to the exact right size to make the countdown full width?

Reply:
Understood your concern, However, In this case, height and width aspect ration will not be maintained by flex in resizable div as well as in normal div too. For that again we have to play with calculated height if we use flex to put it in a single row. Could you enlighten your thoughts on maintaining the height of the canvas in the case should we use flex?

Reply:
Oh ok I was testing on Firefox, forgot that stupid Chrome browser would not do the job correctly (simply adding flex solves all problems on firefox as it forces the canvas on one line, shrink them if necessary and preserve the aspect ratio). It can work with a bit a extra stuff though: add flex on the container element, wrap each canvas in a div then add max-width: 100% on the canvas and it will work on chrome too. Might need to set those wrapping div to flex to to center the canvas vertically though.